Editorial review: Document scroll-triggered animations (#43158)
mdn/content: files/en-us/web/css/reference/properties/animation/index.md

- an optional name for the animation, which may be `none`, a {{cssxref("&lt;custom-ident&gt;")}}, or a {{cssxref("&lt;string&gt;")}}

> [!NOTE]

> > {{cssxref("animation-timeline")}}, {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} are not currently included in this list, as current implementations are reset-only. This means that including `animation` resets a previously-declared `animation-timeline` value to `auto`, previously-declared `animation-range-start` and `animation-range-end` values to `normal`, and a previously-declared `animation-trigger` to `none`, but these properties cannot be set via `animation`. When creating CSS [scroll-driven animations](/en-US/docs/Web/CSS/Guides/Scroll-driven_animations) or [triggered animations](/en-US/docs/Web/CSS/Guides/Animation_triggers), you need to declare these properties after declaring any `animation` shorthand for them to take effect.

Author: Hrm, I tested, and it doesn't seem to reset it. I'll just get rid of that note altogether.

The `animation` property is specified as one or more single animations, separated by commas. Each `animation` within the comma-separated list of animations sets the {{cssxref("animation-name")}}, {{cssxref("animation-duration")}}, {{cssxref("animation-timing-function")}}, {{cssxref("animation-delay")}}, {{cssxref("animation-iteration-count")}}, {{cssxref("animation-direction")}}, {{cssxref("animation-fill-mode")}}, {{cssxref("animation-play-state")}}, and {{cssxref("animation-timeline")}}. If any of the components are not included in an `animation` declaration, the component value is set to the component's initial value.

> [!NOTE]

> > {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} are not currently included in this list, as current implementations are reset-only. This means that including `animation` resets previously-declared `animation-range-start` and `animation-range-end` values to `normal` and a previously-declared `animation-trigger` to `none`, but these properties cannot be set via `animation`. When creating CSS [scroll-driven animations](/en-US/docs/Web/CSS/Guides/Scroll-driven_animations) or [triggered animations](/en-US/docs/Web/CSS/Guides/Animation_triggers), you need to declare these properties after declaring any `animation` shorthand for them to take effect.

Reviewer: I don't think this is accurate, as per https://github.com/mdn/content/pull/43158#discussion_r3059242442, (some values will always be reset only, some are not included, and others are currrently not understood) Also, we need to future proof this. i suggest reworking it to say that the animation resets all the properties, but if a keyword is not understood the entire declaration fails - in does reset it, but that's because it resets every component not declared in the shorthand, so include the longhands after the shorthands.

Author: The link doesn't work, unfortunately, but I do remember it. I ended up deleting a similar note from somewhere else because it didn't seem to be true, at least in the case of `animation-trigger`: I tested it, and the `animation` shorthand doesn't reset `animation-trigger`.

So I'm not really sure what to do for the best. Future-proofing it is also going to be tricky, as they may be added to the shorthand in the future.

For now, I've rewritten it to:

> [!NOTE]
> Other animation-related properties such as {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} cannot be set in the `animation` shorthand, but may be reset by it, depending on the browser implementation. For this reason, it is advisable to declare these longhand properties after declaring any `animation` shorthand to ensure that they take effect.

Any better?

The `animation` property is specified as one or more single animations, separated by commas. Each `animation` within the comma-separated list of animations sets the {{cssxref("animation-name")}}, {{cssxref("animation-duration")}}, {{cssxref("animation-timing-function")}}, {{cssxref("animation-delay")}}, {{cssxref("animation-iteration-count")}}, {{cssxref("animation-direction")}}, {{cssxref("animation-fill-mode")}}, {{cssxref("animation-play-state")}}, and {{cssxref("animation-timeline")}}. If any of the components are not included in an `animation` declaration, the component value is set to the component's initial value.

> ### animation-name

Reviewer: i think we need to put the name back in, but after the note.

Author: Ah. Done.

> > Other animation-related properties such as {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} cannot be set in the `animation` shorthand, but may be reset by it, depending on the browser implementation. For this reason, it is advisable to declare these longhand properties after declaring any `animation` shorthand to ensure that they take effect.

Reviewer: ```suggestion
> The {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} properties are reset by the `animation` shorthand property to their initial values,  `normal`, `normal` and `none`, respectively. These reset-only properties cannot be set in the `animation` shorthand, and including them invalidates the entire declaration. Because the `animation` declaration sets each of these properties to their initial values, they must either be declared after any `animation` shorthand declarations or with greater [specificity](/en-US/docs/Web/CSS/Guides/Cascade/Specificity).
```
 It's not a browser implementation issue. The spec states these three to be reset-only. So, it's not a suggestion... it's a requirement, as browsers that support these properties will (when they follow the spec) reset them. So, we need to make this stronger.

changed from "some properties" to "these three" to ensure people don't think there are some other properties with the same issues. animation-timeline is not reset-only, but not supported in the shorthand by all browsers yet, so that takes a different note. 

If creating a note, you can pull "and including them invalidates the entire declaration." from my suggestion, and add a note with something like "Including invalid values in the `animation` shorthand invalidates the entire declaration. For example, `view()` in a browser that doesn't support `animation-timeline` within the shorthand or including any reset-only property causes the entire declaration to be ignored."

Author: Your suggestion makes sense and brings clarity to this rather confusing issue. Nice one. I've edited it to improve grammar, but will now commit it.

> > The {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} properties are reset by the `animation` shorthand property to their initial values, `normal`, `normal` and `none`, respectively. These reset-only properties cannot be set in the `animation` shorthand, and including them invalidates the entire declaration. Because the `animation` declaration sets each of these properties to their initial values, they must either be declared after any `animation` shorthand declarations or with greater [specificity](/en-US/docs/Web/CSS/Guides/Cascade/Specificity).

Reviewer: ```suggestion
> The `animation` shorthand property resets the {{cssxref("animation-range-start")}}, {{cssxref("animation-range-end")}} and {{cssxref("animation-trigger")}} properties to their initial values; `normal`, `normal` and `none`, respectively. These reset-only properties cannot be set in the `animation` shorthand, and including them invalidates the entire declaration. Because the `animation` declaration sets each of these properties to their initial values, they must either be declared after any `animation` shorthand declarations or with greater [specificity](/en-US/docs/Web/CSS/Guides/Cascade/Specificity).
```
